### ISSUE TITLE
No Images loading - only text.

### ISSUE THREAD

**Issue by SgtSlaughter16** [CommandFusion/iViewer-For-Android#11]
Device: HTC EVO3D 2.3.4 running latest OTA

Have an issue where it doesn't seem any images are being loaded on the device. all I see is text for buttons and such. Even tried just doing a very basic from scratch layout with the new guiDesigner after having the issue using the previous made templates with the DroidViewer Beta and wanted to rule out them being the problem. 

<del>I also noticed the "reload GUI" doesn't really re-sync with the guidesigner when you do it, instead it just refreshes the already loaded gui basically. The only way to reload the gui is to go back to the server address and change it there and thatll trigger it.</del> See #6

And still doing this I can't get any images to load on the device just text from buttons. 

asv file should be here: http://db.tt/GfvtdRxP
gui file should be here: http://db.tt/wOiKxRCu

**Comment by jarrodbell** [comment]
The reload sync issue is a duplicate of #6 it seems.

**Comment by fpillet** [comment]
It seems that redirections are not being properly handled by iViewer on Android yet.

**Comment by SgtSlaughter16** [comment]
Just to double this up tested it out on my HTC EVO4G and had the same issue with it in terms of the images not loading and only showing text.

**Comment by fpillet** [comment]
There was an issue when the URL contained a space (same kind of issue we once had with iViewer on iOS). This has been fixed for build 81.

**Comment by SgtSlaughter16** [comment]
which URL? the one I would put in to sync to had no space in it...

**Comment by fpillet** [comment]
Your DropBox folder had a space in the name, this is what I was seeing but maybe not the core issue with your GUI. It also seems that some of your image names contain a backslash (i.e. the ones in the Themes), it possibly was the issue. Will retest this.

**Comment by SgtSlaughter16** [comment]
just to be clear the files I uploaded to my dropbox were from my computer. I wasn't trying to sync to the files directly in my dropbox. I just didn't have an option to add an attachment. and i just noticed that on this one the asv link I put a "." instead of an "/" incase you needed that file. 

The folder that has all the images and gui/asv files is on my desktop of my laptop all the time.

**Comment by SgtSlaughter16** [comment]
Still showing this issue in build 81 here. used the same ones listed in the original issue and no images loaded.

**Comment by jarrodbell** [comment]
Duplicate of #6

Are the images rendering now? Are you just having the reload issue as mentioned in #6? Or both issues still present?

**Comment by SgtSlaughter16** [comment]
no no images at all are being rendered. just the text still.

**Comment by jarrodbell** [comment]
OK having two issues in here is really getting confusing, as just above you said they load fine first time.
Lets make this issue just for the image loading. Leave the reload refresh stuff for #6

Edited your comments in this issue to pertain only to the issue of images not loading.

**Comment by SgtSlaughter16** [comment]
thanks, yeah when i was saying "they" in terms of the layout. my fault for not being specific there. deff no images loading not even the background still.

**Comment by SgtSlaughter16** [comment]
To update this. I took the SAME layout that i was trying to push to my HTC EVO 3D and pushed it to my wifes kindle fire and......it WORKED! all images loaded up perfectly first time. 

to compare the display specs if it helps any: 
Kindle Fire: 7" multi-touch display with IPS (in-plane switching) technology and anti-reflective treatment, 1024 x 600 pixel resolution at 169 ppi, 16 million colors.
HTC EVO3D: 4.3in qhd 16m colors lcd screen 540 x 960 pixels

The project properties are set at W:540 H:960 by the way.

All done on Milestone 2 also.

**Comment by jarrodbell** [comment]
Which device is failing?
HTC PG86100
HTC PC36100

The info sent through from your registered devices never mentions EVO so not sure what's what.

Im guessing its the lowest OS version device, 2.3.3 or something. Might not be anything we can do about that, old OS are painfully crap. But we'll see.

**Comment by SgtSlaughter16** [comment]
both show the same issue exactly. PG86100 and PC36100 both experience the same issue. The kindle fire is the only one to show the images.

PG86100 is on 2.3.4 and PC36100 is on 2.3.X something too...

**Comment by jarrodbell** [comment]
Both seem to have an older OS than the kindle as well. Is there an upgrade available for your devices by chance? They might have to miss the iViewer cut otherwise.

**Comment by SgtSlaughter16** [comment]
no no upgrade for those devices. the PG86100 is the HTC EVO3D and its next update will be to 4.x...

only reason I bring them up is b/c the support for 2.3.X devices which is what the kindle fire is based off of technically too though a specifically skinned version only Amazon runs as its not even a full google device itself...

so with that said it seems you might have to nix all support for 2.3 devices except the kindle fire?

**Comment by jarrodbell** [comment]
Yeah, the whole android versioning system is a complete mess (mess really isn't a strong enough word here, but lets keep it clean!)

**Comment by SgtSlaughter16** [comment]
well it gets messy when you have companies like amazon take it and strip everything out making it "not android" at all which makes it 100x more difficult for devs. This issue is odd b/c it wasn't on the last beta, but anything possible with it being re-written from ground up...

**Comment by jarrodbell** [comment]
Yeah the last beta was simply not anywhere near iViewer quality, and was scrapped completely. You cannot compare anything to it as it is a complete rewrite and drawing methods it used would have resulted in a lot of sacrifices in other areas.

**Comment by SgtSlaughter16** [comment]
Well I have full root access to my phone's running 2.3.X if any kind of log's of anything could be valuable just let me know....Unfortunately it sounds more like you might just end up making it only supported in 4.0...?

Im still trying to figure out why the KindleFire was able to pull the images fine and my other devices that run actual official Android were not able to....

I did think it was odd that the display on the KindleFire(1024x600) was completely filled out by the background image which was set to the resolution of my phone which was (960x540)....now as I stated before my phone is only 4.3in screen size and the kindlefire is 7in....could that somehow have something to do with it??

**Comment by fpillet** [comment]
Will check this out. One important thing to note is that our only real 2.x test devices here are Kindle Fire units, which explains why iViewer works fine on them. The problem with Android is that each device has its own issues, and we have to go through painful testing on each device (which we obviously can't do, so we rely on beta testing to get a wider range of devices tested).

It's possible that your issue is a trivial one we completely missed. We'll double check that during the coming week.

Florent

**Comment by SgtSlaughter16** [comment]
Okay. completely understand. like I said if need be I can pull all and any kind of log's from my device if you even need me to personally test some things out just let me know.

**Comment by fpillet** [comment]
Getting back to this issue now that I've got some other crucial problems solved. I need your full GUI including images, can you zip it and put it somewhere in your dropbox? Thanks!

**Comment by SgtSlaughter16** [comment]
This should be it.
I did yet another from complete scratch and moved the 3 images being used(button and background) to a new workspace folder to separate all past stuff and here is what it is now.

http://db.tt/G0yeQxej

same result as before. only white text of button names load image wise on a black background.

**Comment by fpillet** [comment]
Your latest GUI works fine here, please let me know how latest build fares regarding this issue. If it's still failing, ping me on IRC or Skype, we'll need to do some deeper debugging.

**Comment by SgtSlaughter16** [comment]
Okay. Ill likely get to doing it fully tested and have time to discuss on Monday, should there be any problems...

**Comment by fpillet** [comment]
Excellent. We're trying to get most issues fixed and the software generally stabilized before carrying on with the remaining missing features (page transitions, video support, debugger interface)

**Comment by SgtSlaughter16** [comment]
I wanted to test some more things out before getting in chat exactly since I'm still having the issue here as of build 83 with background images not loading. 

I went into GUI designer and set the background a as a solid color(have a bug on doing this as well at the end) and the solid color loaded just fine on my device. The button images still do not load but the color set for the background loaded fine. I also tried other images that were smaller than the one I was trying before and still couldn't get them to load.(even a 100x69 jpeg image wouldn't load...  

Note: setting a background to a solid color in GUI Designer has a lil bug. When you select the color in the "Background Theme Properties box" you are shown a grey box still where it previews your background. Even clicking OK to make the background and then saving the whole Project and then bringing back up the background theme properties box results in the same grey box. Its not until you exit GUI designer and bring it back up that the background theme properties box will show the preview as the color selected. Then it can be properly applied to the project page and shown on the large project preview.
Also noticed that even though I select an image to be tiled and it shows as tiled in the preview part of "background theme properties" it still is not tiled when viewing the main preview of the project.
